submissions/OlStaniev/friends-app/script.js

```diff
@@ -0,0 +1,155 @@
+const form = document.querySelector('.searchForm')
+const searchIput = document.querySelector('#searchName')
+const burgerMenu = document.querySelector('.burgerForForm')
+const paginationList = document.querySelector('.paginationList')
+
+let users = []
+const numberUsersOnPage = 24
+
+document.addEventListener("DOMContentLoaded", loadData)
+form.addEventListener('input', () => renderPageContent(1))
+form.addEventListener("submit", (e) => e.preventDefault())
+burgerMenu.addEventListener('click', searchFormSwitch)
+paginationList.addEventListener('click', changePage)
+
+
+async function loadData() {
+    try {
+        const response = await fetch(url)
+        if (!response.ok) {
+            throw new Error(`http status ${response.status}`)
+        }
+        const data = await response.json()
+        users = data.results
+        renderPageContent(1)
+    }
+    catch (e) {
+        console.error(`FRIENDS ERROR: ${e}`)
+        let secondsToReload = 5
+        setInterval(() => {
+            cardsField.innerHTML =
+                `<div class="error">
+                <p>Somthing went wrong</p>
+                <p>The page will reload in ${secondsToReload} sec</p>
+            </div>`
+            if (--secondsToReload === -1) { location.reload() }
+        }, 1000)
+    }
+}
+function renderPageContent(currentPage) {
+    const usersForRender = prepairUsersForRender()
+    createPaginationList(usersForRender, currentPage)
+    renderCards(preparCardsToRender(getUsersForCurrentPage(usersForRender, currentPage)))
+}
+function prepairUsersForRender() {
+    const searchedByName = searchByName(users)
+    const sortedByAge = sortingUsersByAge(searchedByName)
+    const sortedByAlphabet = sortingUsersByAlphabet(sortedByAge)
+    const filteredByGender = filterByGender(sortedByAlphabet)
+    return filteredByGender
+}
+function searchByName(users) {
+    if (form.searchName.value) {
+        return users.filter(user =>
+            (user.name.first + ' ' + user.name.last).toLowerCase()
+                .includes(form.searchName.value.toLowerCase()))
+    }
+    return users
+}
+function sortingUsersByAge(users) {
+    if (form.sort.value === 'ageUp') {
+        return users.sort((userA, userB) =>
+            userA.dob.age < userB.dob.age ? -1 : 1)
+    }
+    if (form.sort.value === 'ageDown') {
+        return users.sort((userA, userB) =>
+            userA.dob.age < userB.dob.age ? 1 : -1)
+    }
+    return users
+}
+function sortingUsersByAlphabet(users) {
+    if (form.sort.value === 'az') {
+        return users.sort((userA, userB) =>
+            userA.name.first < userB.name.first ? -1 : 1)
+    } if (form.sort.value === 'za') {
+        return users.sort((userA, userB) =>
+            userA.name.first > userB.name.first ? 1 : -1)
+    }
+    return users
+}
+function filterByGender(users) {
+    return users.filter(user =>
+        user.gender === form.gender.value || form.gender.value === 'all'
+    )
```

Comment:
No need in running filtration if you need to show all items

Reply:
+